### ISSUE TITLE
Feature Request: LangSmith integration

### ISSUE THREAD

**Issue by parth-savjani** [feder-cr/linkedIn_auto_jobs_applier_with_AI#228]
LangSmith integration will help,

1. Removing manual logging (API calls, cost calculation, etc)
2. No need for creating incremental log files on the local system
3. Tech/Non-tech users can get an interactive dashboard as shown in the attachment. 

Possible downside
1. Dependancy on the Langchain API key, need to get an additional key and add it to the secret.yaml file
2. Data Privacy issue as we are sharing personal data 

Thanks!
![Screenshot 2024-09-01 at 12 59 17 PM](https://github.com/user-attachments/assets/cca262a2-b34a-458c-ae08-40e4d9c86f59)

**Comment by timeowilliams** [comment]
@parth-savjani , I like this idea. Have you run into any costs so far with using langsmith? 

If users move from using OpenAI to a local llama3 server for AI, will this still work with no issues?

**Comment by parth-savjani** [comment]
Yes, I have been running it with l lama3.1 and Lang Smith, zero
infrastructure cost provided one can run llama3.1 locally. I am running it
on my MacBook pro with m1.

However, I have kept both openAI and llama3.1 options in codebase.

Only difference I found is sometimes llama doesn't follow the prompts
properly and generate longer response instead of single value response
which can be filled as answer to additional question of easy application.
